### PR TITLE
Separate requirements.txt file for ORTModule pipelines

### PR DESCRIPTION
**Pull request by baijumeswani**
Separate out the dependencies of ```ORTModule``` into its own ```requirements.txt``` file that will facilitate the merger of dev branch into the master branch.

### CODE REVIEW

**Review by thiagocrepaldi — APPROVED** [review]
I would consider another name as opposed to `secondary`, but it is good to have this separation

**Comment by baijumeswani** [comment]
Maybe, I can just call the folder ```deepspeed``` and can be referenced as ```deepspeed/requirements.txt```?

File: tools/ci_build/github/linux/docker/scripts/training/requirements.txt [hunk 1]
```diff
@@ -1,15 +1,11 @@
 --pre
--f https://download.pytorch.org/whl/nightly/cu101/torch_nightly.html
+-f https://download.pytorch.org/whl/torch_stable.html
```
Comment (thiagocrepaldi) [inline tools/ci_build/github/linux/docker/scripts/training/requirements.txt:2]: why pytorch 1.6?

File: tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+--pre
+-f https://download.pytorch.org/whl/nightly/cu101/torch_nightly.html
+# transformers requires sklearn
+pandas
+sklearn
+numpy==1.19.5
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements.txt:6]: If you use numpy 1.19, then the built package can't be used with numpy 1.16 and basically it is not distributable. It's fine if it is only used for CI build testing. But when your team decided to publish the package, you need to change this part.

Comment (baijumeswani) [reply tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements.txt:6]: This change is only for the CI build. It does not impact the packaging pipeline and should not impact our distribution.

File: orttraining/orttraining/test/python/how_to_install_training_test_dependencies.md [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
-Due to a [bug on DeepSpeed](https://github.com/microsoft/DeepSpeed/issues/663), we need a two step process for installing dependencies ORTModule dependencies.
+Due to a [bug on DeepSpeed](https://github.com/microsoft/DeepSpeed/issues/663), we need a two step process for installing ORTModule dependencies.
 
 1. Install first set of dependencies:
 
    ```sh
-   pip install -r tools/ci_build/github/linux/docker/scripts/training/requirements.txt
+   pip install -r tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements.txt
 
 2. Install second set of dependencies for ortmodule:
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/how_to_install_training_test_dependencies.md:13]: clarify *why* a second set is needed

Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/how_to_install_training_test_dependencies.md:13]: # Due to a [bug on DeepSpeed](https://github.com/microsoft/DeepSpeed/issues/663), we install it separately through ortmodule/stage2/requirements.txt

Comment (baijumeswani) [reply orttraining/orttraining/test/python/how_to_install_training_test_dependencies.md:13]: There is a comment on that earlier in this file that addresses this (line6).

**Comment by baijumeswani** [comment]
Thank you for the review.

**Pull request merged**